apps/backoffice-v2/src/lib/blocks/hooks/useEntitiesDocuments/useEntitiesDocuments.ts

```diff
@@ -0,0 +1,119 @@
+import { useDocumentsAdapter } from '@/domains/documents/hooks/useDocumentsAdapter/useDocumentsAdapter';
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { selectDirectors } from '@/pages/Entity/selectors/selectDirectors';
+import { TDocument } from '@ballerine/common';
+import { useCallback, useMemo } from 'react';
+
+export type TEntitiesDocumentsFilter = 'all' | 'directors' | 'ubos';
+
+const allOrExact = (filter: TEntitiesDocumentsFilter, exact: TEntitiesDocumentsFilter) =>
+  filter === 'all' || filter === exact;
+
+export const useEntitiesDocuments = (
+  workflow?: TWorkflowById,
+  filter: 'all' | 'directors' | 'ubos' = 'all',
+) => {
+  const directors = useMemo(
```

Comment:
We need to stop with those huge hooks, hook is a framework utility it shouldnt have this amount of logic - logic should be a pure function stored in a regular ts file

hook should be a thin layer that use business logic functions

Reply:
see slack